### DISCUSSION TITLE
coordinates problem with example UpdateArea.ino

### DISCUSSION THREAD

**Discussion by steff2000** [olikraus/u8g2#1597] [Q&A] [answered]
I'm really new with u8g2.
I use the c-lib with an ESP32 and the esp-idf framework.
My display is a cheap 128x64 SSD1306 driven through I2C.

I initialized the u8g2 lib with

u8g2_Setup_ssd1306_i2c_128x64_noname_f(...);

The library seems to work fine!

Using the arduino example from
https://github.com/olikraus/u8g2/blob/master/sys/arduino/u8g2_full_buffer/UpdateArea/UpdateArea.ino
I stumbled over a 'coordinates' problem. Of course I have changed all C++ calls to the corresponding C calls.

The line (325)

u8g2_DrawStr(&u8g2, 0, 12,"UpdateDisplayArea");

leaves space at the top of the display. The string is y shifted downwards.

I changed it to
u8g2_DrawStr(&u8g2, 0, 0,"UpdateDisplayArea");
and now it looks fine and uses the topmost position of the display.

But this is contrary to the FAQ?
- Do not output a text at (0,0): The reference point for text is lower left,
  so u8g2_DrawStr(&u8g2, 0, 0, “Hello world!”); will not display anything.
	Instead use u8g2_DrawStr(&u8g2, 0, 20, “Hello world!”);

Same example next problem:

The "u8g2" String will be drawn into the buffer by

Line 345
	// draw the scolling text
		u8g2_DrawUTF8(&u8g2,
									pixel_area_x_pos - width + offset, 
									pixel_area_y_pos + pixel_area_height +
									u8g2_GetDescent(&u8g2) - 1, 
									text);

The string 'u8g2' is clearly shifted downwards in the frame.

I have fixed the y shift by subtracting '16'.
	// draw the scolling text
		u8g2_DrawUTF8(&u8g2,
									pixel_area_x_pos - width + offset, 
									pixel_area_y_pos - 16 + pixel_area_height +
									u8g2_GetDescent(&u8g2) - 1, 
									text);

Now it looks perfect.

Can someone explain the y shifting?

**Comment by olikraus** [comment, marked as answer]
Maybe https://github.com/olikraus/u8g2/wiki/u8g2reference#setfontpostop is active. The documentation usually assumes https://github.com/olikraus/u8g2/wiki/u8g2reference#setfontposbaseline

**Reply by steff2000** [reply]
Ah, YES, it was my fault. Now it´s cyrstal clear! 
Thank you for writing u8g2!
